reframe/frontend/executors/policies.py

```diff
@@ -55,6 +59,21 @@ def runcase(self, case):
         except BaseException:
             task.fail(sys.exc_info())
         finally:
+
+            # Execute cleanup of dependencies if all dependent cases have been
+            # executed
+            for dep in self.dependency_tree[case]:
+                self.dependency_count[dep] -= 1
+                if self.dependency_count[dep] == 0:
+                    # Check if dep has failed before cleaning
+                    for t in self.stats.tasks():
+                        if t.testcase == dep and t.failed is False:
+                            dependency_clean_up_task = RegressionTask(dep)
```

Comment:
Why do you create a new "task" here? I'd just call the deferred `cleanup()` on the original one.

Also cleanup may throw, so it has to be treated correctly. That's why I don't think putting this loop in the `finally` is correct.

Reply:
Done

Reply:
Done